### PR TITLE
fix: ensures autosaves run sequentially rather than ever in parallel

### PR DESCRIPTION
**Pull request by jmikrut**
With a small autosave interval, it's possible for two autosaves to be executing in parallel. 

This PR ensures that you can fire many autosaves, but they will be executed sequentially rather than ever being able to be run in parallel.

### CODE REVIEW

File: packages/ui/src/elements/Autosave/index.tsx [hunk 1]
```diff
@@ -88,6 +91,32 @@ export const Autosave: React.FC<Props> = ({ id, collection, global: globalDoc })
...
+    } catch (error) {
+      console.error('Error autosaving:', error)
+    } finally {
+      setQueue((prevQueue) => prevQueue.slice(1))
+      setIsProcessing(false)
+    }
+  }, [isProcessing, queue])
+
+  // Process the queue whenever it changes
+  useEffect(() => {
```
Comment (JarrodMFlesch) [inline packages/ui/src/elements/Autosave/index.tsx:114]: Can we do this without a useEffect? Seems like it should be possible

Comment (jmikrut) [reply packages/ui/src/elements/Autosave/index.tsx:111]: I would love to learn

Comment (JarrodMFlesch) [reply packages/ui/src/elements/Autosave/index.tsx:111]: Maybe it cant, I hate it but will live

Comment (JarrodMFlesch) [reply packages/ui/src/elements/Autosave/index.tsx:111]: Can't you just call `processCurrentAutosave` and have it take the autosave function as an arg and leave everything else as it is? Removing the autosave state and the useEffect?

**Comment by JarrodMFlesch** [comment]
Should update the description here, I think your initial commit allow them to fire sequentially but now it just allows 1 to run and future autosaves cannot be called if the prev one has not finished.

**Comment by jmikrut** [comment]
Closing in favor of #9892

**Pull request closed without merging**